workload/report/schema.py

```diff
@@ -0,0 +1,460 @@
+from enum import StrEnum, auto
+import json
+from typing import Optional, Any
+
+from pydantic import BaseModel, model_validator
+import yaml
+
+
+class Parallelism(BaseModel):
+    """Accelerator parallelism details."""
+
+    dp: int = 1
+    """Data parallelism level."""
+    tp: int = 1
+    """Tensor parallelism level."""
+    pp: int = 1
+    """Pipeline parallelism level."""
+    ep: int = 1
+    """Expert parallelism level."""
+
+
+class HostAccelerator(BaseModel):
+    """Host accelerator details."""
+
+    model: str
+    """Accelerator model."""
+    memory: float | int
+    """Amount of memory in one accelerator, in GB."""
+    count: int
+    """Number of accelerators."""
+    parallelism: Optional[Parallelism] = None
+    """Parallelism configuration used."""
+    metadata: Optional[Any] = None
+
+
+class Host(BaseModel):
+    """Host hardware details."""
+
+    accelerator: list[HostAccelerator]
+    metadata: Optional[Any] = None
+
+
+class EngineDetails(BaseModel):
+    """Inverence engine details."""
+
+    name: str
+    version: Optional[str] = None
+    args: dict[str, Any]
+    metadata: Optional[Any] = None
+
+
+class Platform(BaseModel):
+    """Software platform details endompassing all inference engines."""
+
+    engine: list[EngineDetails]
+    """Details on inference engines, list corresponds 1:1 with scenario.host.accelerator."""
+    metadata: Optional[Any] = None
+
+
+class Model(BaseModel):
+    """AI model details."""
+    name: str
+    quantization: Optional[str] = None
+    adapters: Optional[list[dict[str, str]]] = None
+    metadata: Optional[Any] = None
+
+
+class WorkloadGenerator(StrEnum):
+    """
+    Enumeration of supported workload generators
+
+    Attributes
+        FMPERF: str
+            fmperf
+        GUIDELLM: str
+            GuideLLM
+        INFERENCE_PERF: str
+            Inference Perf
+        VLLM_BENCHMARK: str
+            benchmark_serving from vLLM
+    """
+
+    FMPERF = auto()
+    GUIDELLM = auto()
+    INFERENCE_PERF = 'inference-perf'
+    VLLM_BENCHMARK = 'vllm-benchmark'
+
+
+class Load(BaseModel):
+    """Workload for benchmark run."""
+
+    name: WorkloadGenerator
+    """Workload generator"""
+    type: Optional[str] = None
+    args: Optional[dict[str, Any]] = None
+    metadata: Optional[Any] = None
+
+
+class Scenario(BaseModel):
+    """System configuration and workload details for benchmark run."""
+
+    description: Optional[str] = None
+    host: Optional[Host] = None
```

Comment:
Do we want to provide a way to specify multiple hosts as well when multiple replicas are run?

Reply:
Yes, we need multiple hosts.